src/ntp_synchronizer/ntp_synchronizer.go

```diff
@@ -0,0 +1,162 @@
+	return util.Execute(command, args...)
+}
+
+func convertSourceState(val string) models.SourceState {
+	switch val {
+	case "*":
+		return models.SourceStateSynced
+	case "+":
+		return models.SourceStateCombined
+	case "-":
+		return models.SourceStateNotCombined
+	case "?":
+		return models.SourceStateUnreachable
+	case "x":
+		return models.SourceStateError
+	case "~":
+		return models.SourceStateVariable
+	default:
+		return models.SourceStateError
+	}
+}
+
+func AddServer(e NtpSynchronizerDependencies, ntpSource string) error {
+	stdout, stderr, exitCode := e.Execute("chronyc", "add", "server", ntpSource, "iburst")
+
+	if exitCode == 0 {
+		return nil
+	} else {
+		return errors.Errorf("chronyc exited with non-zero exit code %d: %s\n%s", exitCode, stdout, stderr)
+	}
+}
+
+func formatChronySourcesOutput(output string) []*models.NtpSource {
+	sources := make([]*models.NtpSource, 0)
+
+	for _, line := range strings.Split(output, "\n") {
+		// Skip empty lines
+		if line == "" {
+			continue
+		}
+
+		fields := strings.Fields(line)
+
+		// Skip whatever is not a server
+		if len(fields) == 0 || fields[0][0] != '^' {
+			continue
+		}
+
+		sources = append(sources, &models.NtpSource{SourceName: fields[1], SourceState: convertSourceState(string(fields[0][1]))})
+	}
+
+	return sources
+}
+
+func GetNTPSources(e NtpSynchronizerDependencies) ([]*models.NtpSource, error) {
+	stdout, stderr, exitCode := e.Execute("timeout", strconv.FormatInt(ChronyTimeoutSeconds, 10), "chronyc", "-n", "sources")
+
+	switch exitCode {
+	case 0:
+		return formatChronySourcesOutput(stdout), nil
+	case util.TimeoutExitCode:
+		return nil, errors.Errorf("chronyc was timed out after %d seconds", ChronyTimeoutSeconds)
+	default:
+		return nil, errors.Errorf("chronyc exited with non-zero exit code %d: %s\n%s", exitCode, stdout, stderr)
+	}
+}
+
+func isServerConfigured(executer NtpSynchronizerDependencies, server string) (bool, error) {
+	// Check if the server is one of the configured sources
+	sources, err := GetNTPSources(executer)
+
+	if err != nil {
+		return false, errors.Wrapf(err, "Failed to get NTP sources")
+	}
+
+	for _, source := range sources {
+		if server == source.SourceName {
+			return true, nil
+		}
+	}
+
+	// Check if one of the server CNames is one of the configured sources
+	names, err := net.LookupHost(server)
+
+	if err != nil {
+		return false, errors.Wrapf(err, "Failed to lookup server %s", server)
+	}
+
+	for _, source := range sources {
+		if funk.Contains(names, source.SourceName) {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+func NtpSync(ntpSyncRequestStr string, executer NtpSynchronizerDependencies, log logrus.FieldLogger) (stdout string, stderr string, exitCode int) {
+	var ntpSyncRequest models.NtpSynchronizationRequest
+
+	err := json.Unmarshal([]byte(ntpSyncRequestStr), &ntpSyncRequest)
+	if err != nil {
+		log.WithError(err).Errorf("NTPSync: json.Unmarshal")
```

Comment:
Can these be more like sentences? Maybe something like:

```suggestion
		log.WithError(err).Errorf("Failed to unmarshal ntp request string %s", ntpSyncRequestStr)
```

Same goes for pretty much every error message in this function